scala-thoughts.md

```diff
@@ -0,0 +1,9 @@
+Well, that may have been the most frustrating coding experience I've had since I learned to code. Wrestling my mind into a functional programming mode (which I haven't done in 3 semesters) was challenging in itself, but the most difficult thing about this homework was dealing with SyntaxErrors and TypeErrors. Of the ~12 hours I spent on the assignment I feel like I spent maybe 3 hours designing the actual algorithms to solve the problems. While I guess it's nice that ScalaIDE tells you about TypeErrors before compile time or even runtime, their error messages are truly obfuscating and don't point you anywhere near the right direction (which is usually that I forgot to define the return type of a function (so it defaulted to unit) or that I forgot an equals sign when defining a function).
+
```

Comment:
I also found the scaled errors really distracting. I wish there was a way to stop them from showing up until runtime or at least until you stop typing!